improvement: optimize Tamagui component import patterns

huhuanming opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added cross-platform Anchor links, a theme-aware Spinner, Fieldset with horizontal variant, TamaguiProvider (with Portal), new TextArea/Input components and size-variant helpers.

* **Refactor**
  * Consolidated UI primitives, hooks and types behind a single internal shared Tamagui surface; imports redirected with no behavioral change.

* **Chores**
  * Type/import cleanups, added centralized web font-family constant, and normalized native build TAMAGUI_TARGET.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

huhuanming committed:
Update SimpleDb.ts

huhuanming committed:
Merge branch 'x' into x-startup

huhuanming committed:
fix: fix tamagui

huhuanming committed:
Update type.ts

huhuanming committed:
fix: fix lint

huhuanming committed:
Update index.tsx

huhuanming committed:
Update index.tsx

huhuanming committed:
Update list.tsx

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/e377e4ef-093e-41e8-8097-459a7e5b41b3?refs=7fa18bae65176ef4a76be9907f04de8e251a4b47&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/e377e4ef-093e-41e8-8097-459a7e5b41b3/license?refs=7fa18bae65176ef4a76be9907f04de8e251a4b47&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

sidmorizon commented on packages/components/src/actions/ActionList/index.tsx:
```diff
@@ -1,10 +1,11 @@
 import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 
+import { withStaticProperties } from '@tamagui/helpers';
+import { useMedia } from '@tamagui/web';
```
在 shared 里弄一个 tamagui.ts 来导出这些用到的组件吧，以后统一走 shared 导入，防止未来又直接从 tamagui import

用 eslint 禁止业务直接 import tamagui，必须从 shared 里 import， tamagui.ts 内部 import 时用 eslint disable rule 规避检查即可

huhuanming replied:
okay

huhuanming committed:
Merge branch 'x' into x-startup

huhuanming committed:
Refactor web font family usage and config loading

Moved the web font family string to a shared utility file and updated all references to use it. Changed Tamagui config loading in Provider to use dynamic require and type annotation. Minor type additions and platformEnv usage update for native detection.

huhuanming committed:
Merge branch 'x-startup' of github.com:OneKeyHQ/app-monorepo into x-startup

huhuanming committed:
Set TAMAGUI_TARGET to 'native' by default in mobile config

Added logic to ensure process.env.TAMAGUI_TARGET is set to 'native' in the mobile Babel config if not already defined. Also fixed import order in tamagui.config.ts for consistency.

huhuanming committed:
Refactor Tamagui imports to shared module

Replaces all Tamagui-related imports from external packages (e.g., '@tamagui/web', '@tamagui/core', '@tamagui/text', etc.) with imports from a new local shared module '../../shared/tamagui'. This centralizes Tamagui dependencies, simplifies import paths, and improves maintainability.

huhuanming committed:
fix: fix lint

huhuanming committed:
fix: fix lint

huhuanming committed:
Update type.ts

huhuanming committed:
fix: fix lint

huhuanming committed:
Merge branch 'x' into x-startup

huhuanming committed:
Update index.tsx

huhuanming committed:
Update tamagui.config.ts

huhuanming committed:
Update tamagui.config.ts

huhuanming committed:
Update tamagui.config.ts

huhuanming committed:
Update tamagui.config.ts

huhuanming committed:
Update babel.config.js

huhuanming committed:
Update babel.config.js

huhuanming committed:
Update .folderslintrc
